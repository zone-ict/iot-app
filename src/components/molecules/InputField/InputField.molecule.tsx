import React, { DetailedHTMLProps, InputHTMLAttributes } from 'react';
import InputText from '../../atoms/InputText/InputText.atom';

interface Props extends DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement> {
  label?: string;
  error?: string;
}

export default function InputField({ label, error, ...props }: Props) {
  return (
    <div>
      <h6>{label}</h6>
      {/* Disabled prop spreading rule because we're making a generic component that'll 
        accept all manner of props attributed for input element
      */}
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <InputText {...props} />
      {error && <div>{error}</div>}
    </div>
  );
}
