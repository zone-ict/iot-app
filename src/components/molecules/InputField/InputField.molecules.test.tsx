import React from 'react';
import renderer from 'react-test-renderer';
import InputField from './InputField.molecule';

describe('Input field component', () => {
  it('should be able to render correctly without any props', () => {
    const tree = renderer.create(<InputField />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to render correctly when passed label and error props', () => {
    const tree = renderer.create(<InputField label="test" error="test" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to render correctly with only label props passed', () => {
    const tree = renderer.create(<InputField label="test" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to render with only error prop passed', () => {
    const tree = renderer.create(<InputField error="test" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
