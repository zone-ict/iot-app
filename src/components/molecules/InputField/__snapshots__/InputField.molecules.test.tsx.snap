// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field component should be able to render correctly when passed label and error props 1`] = `
<div>
  <h6>
    test
  </h6>
  <input
    className="css-1fq7r63"
  />
  <div>
    test
  </div>
</div>
`;

exports[`Input field component should be able to render correctly with only label props passed 1`] = `
<div>
  <h6>
    test
  </h6>
  <input
    className="css-1fq7r63"
  />
</div>
`;

exports[`Input field component should be able to render correctly without any props 1`] = `
<div>
  <h6 />
  <input
    className="css-1fq7r63"
  />
</div>
`;

exports[`Input field component should be able to render with only error prop passed 1`] = `
<div>
  <h6 />
  <input
    className="css-1fq7r63"
  />
  <div>
    test
  </div>
</div>
`;
